src/lib/fragments/Accordion.react.js

```diff
@@ -50,39 +212,58 @@ class Accordion extends Component {
 
   renderAccordionSection(index, accordionSectionContent, sectionIsOpen) {
     const accordionHeading = this.props.accordionHeadings[index]
-      return (
-          <div className={sectionIsOpen ? "govuk-accordion__section govuk-accordion__section--expanded" : "govuk-accordion__section"}>
-            <div className="govuk-accordion__section-header">
-              <h2 className="govuk-accordion__section-heading">
-                <button className="accordion-button govuk-accordion__section-button" type="button" aria-controls={`accordion-default-content-${index}`} aria-expanded={sectionIsOpen} onClick={() => this.openOrCloseAccordionSection(index)}>
-                  <span className="govuk-accordion__section-heading-text" >
-                    <span className="govuk-accordion__section-heading-text-focus"> {accordionHeading} 
-                    </span>
-                  </span> 
-                  <span className="govuk-visually-hidden govuk-accordion__section-heading-divider"></span>
-                  <span className="govuk-accordion__section-toggle" data-nosnippet>
-                    <span className="govuk-accordion__section-toggle-focus">
-                      <span className="moj-side-navigation__item--collapsed">
-                      </span>
-                    </span>
-                    <span className= {sectionIsOpen ?  "govuk-accordion-nav__chevron" : "govuk-accordion-nav__chevron govuk-accordion-nav__chevron--down"} ></span>
-                    <span className="govuk-accordion__section-toggle-focus govuk-accordion__section-toggle-text"> {sectionIsOpen ? "Hide" : "Show"} 
-                      <span className="govuk-visually-hidden"> this section</span>
-                    </span>
+    const contentId = `accordion-default-content-${index}`;
+    return (
+      <div
+        className={sectionIsOpen ? "govuk-accordion__section govuk-accordion__section--expanded" : "govuk-accordion__section"}
+        data-section-index={index}
+      >
+        <div className="govuk-accordion__section-header">
+          <h2 className="govuk-accordion__section-heading">
+            <button
+              className="accordion-button govuk-accordion__section-button"
+              type="button"
+              aria-controls={contentId}
+              aria-label={
+                sectionIsOpen
+                  ? `Heading at level ${index} is ${accordionHeading},,,, Section is open, select to close`
+                  : `Heading at level ${index} is ${accordionHeading},,,, Section is closed, select to open`
```

Comment:
capitalisation of "section" and "select" as above. Also, why the commas?

Reply:
I believe this was so there was a gap between heading at level and section is open when being read out by the screen reader

Reply:
Sounds good - thanks for explaining